content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-labels/index.md

```diff
@@ -0,0 +1,869 @@
+
+Don't like the inline styling of the labels? No problem. You can style `<label>` elements like any other. In this case, we'll leverage the previously used flex styling to make the labels appear above the inputs:
+
+```html
+<form style="display: flex; gap: 1rem;">
+	<label style="display: flex; flex-direction: column;">
+        <span>Username</span>
+        <input name="username" type="text"/>
+	</label>
+	<label style="display: flex; flex-direction: column;">
+        <span>Password</span>
+	    <input name="password" type="password"/>
+    </label>
+    <div style="display: flex; flex-direction: column; justify-content: flex-end;">
+        <button type="submit">Login</button>
+    </div>
+</form>
+```
+
+----
+
+<form style="display: flex; gap: 1rem;" onsubmit="event.preventDefault()">
+	<label style="display: flex; flex-direction: column;">
+        <span>Username</span>
+        <input name="username" type="text"/>
+	</label>
+	<label style="display: flex; flex-direction: column;">
+        <span>Password</span>
+	    <input name="password" type="password"/>
+    </label>
+    <div style="display: flex; flex-direction: column; justify-content: flex-end;">
+        <button type="submit">Login</button>
+    </div>
+</form>	
+
+
+---
+
+It's worth noting, however, that while `<span>` tags are absolutely welcomed and allowed to be used in a `<label>`, there are some restrictions on the elements you can have as child tags.
+
+Only tags that are considered [phrasing content](https://developer.mozilla.org/en-US/docs/Web/HTML/Guides/Content_categories#phrasing_content) are allowed; otherwise a screen-reader may not properly handle the text inside.
+
+This mean that while **the following incomplete list of elements are allowed**:
+- `<b>`
+- `<em>`
+- `<i>`
+- `<span>`
+- `<strong>`
+
+**The rest of HTML elements are not allowed**, like:
+
+- `<p>`
+- `<h1>` through `<h6>`s
+- `<article>`s, `<nav>`s, and other landmark elements
+- Other `<label>`s
+- Any other `<input>` elements than the one you want labelled
+  - One label per input only!
+
+
+# Why You Shouldn't Use Placeholders
+
+Whenever the topic of element association comes up, I regularly get asked the following:
+
+> Why don't you just use placeholders in an element?
+
+It's a valid question, given that it's been adopted as a broadly utilized pattern for many forms in recent years. Additionally — at least visually — it seems like placeholders provide a similar level of information as labels might.
+
+---
+
+
+<form style="display: flex; gap: 1rem;" aria-hidden="true" onsubmit="event.preventDefault()">
+    <input placeholder="Username" name="username" type="text"/>
+    <input placeholder="password" type="password"/>
+    <button type="submit">Login</button>
+</form>
+
+
+----
+
+Despite their popularity, **placeholders have been widely seen as a harmful U.X. pattern for inputs by accessibility experts**. Some of the issues with placeholders these experts cite are:
```

Comment:
Consider including that hints cannot be seen when the form is filled out, so longer forms are harder to validate without deleting the contents of the input